internal/rt/fastmem.go

```diff
@@ -108,14 +108,14 @@ func GuardSlice(buf *[]byte, n int) {
 }
 
 func AddPtr(a unsafe.Pointer, b uintptr) unsafe.Pointer {
-	return unsafe.Pointer(uintptr(a) + b)
+	return unsafe.Add(a, b)
 }
 
 func SubPtr(a unsafe.Pointer, b uintptr) unsafe.Pointer {
-	return unsafe.Pointer(uintptr(a) - b)
+	return unsafe.Add(a, -b)
 }
 
-func PtrOffset(a unsafe.Pointer, b unsafe.Pointer) int {
+func PtrOffset(a uintptr, b uintptr) int {
 	return int(uintptr(a)) - int(uintptr(b))
```

Comment:
Why did you change the arguments from unsafe.Pointer to uintptr?
And if you use uintptr, why don't you just return `int(a - b)`?

Reply:
in case of bad pointer pass to this func. Actually can pass `int`, just too lazy to change old codes